creator-node/src/middlewares.js

```diff
@@ -186,6 +187,118 @@ async function triggerSecondarySyncs (req) {
   }
 }
 
+/**
+ * Trigger SyncRequests to both secondaries, and wait for at least one to sync before returning
+ */
+async function triggerAndWaitForSecondarySyncs (req, { primaryClockVal, immediate, enforceQuorum }) {
+  const serviceRegistry = req.app.get('serviceRegistry')
+  const { snapbackSM } = serviceRegistry
+
+  if (config.get('isUserMetadataNode') || config.get('snapbackDevModeEnabled')) {
+    req.logger.info('TODO msg')
+    return
+  }
+
+  if (!req.session || !req.session.wallet) {
+    req.logger.error(`TriggerSecondarySyncsNew Error - req.session.wallet missing`)
+    return
+  }
+  const wallet = req.session.wallet
+
+  try {
+    if (!req.session.nodeIsPrimary || !req.session.creatorNodeEndpoints || !Array.isArray(req.session.creatorNodeEndpoints)) {
+      req.logger.info('TODO')
+      return
+    }
+
+    let [primary, ...secondaries] = req.session.creatorNodeEndpoints
+    secondaries = secondaries.filter(secondary => (!!secondary && _isFQDN(secondary)))
+
+    if (primary !== config.get('creatorNodeEndpoint')) {
+      throw new Error('TODO')
+    }
+
+    const timeoutMs = enforceQuorum ? 20000 : 5000
```

Comment:
Why have this?

Reply:
discussed offline, will instead have a default value here but have client pass in desired timeout